AH rewrite: Add function to compute coords of new fast flow iteration

knelli2 opened the pull request:
## Proposed changes

Seventh PR in horizon finder changes. Adds a function that computes the block logical coordinates of the current iteration strahlkorper. Retries the computation with an adjusted strahlkorper if it fails. Also handles extrapolation for the zeroth fast-flow iteration surface.

<!--
At a high level, describe what this PR does.
-->

### Upgrade instructions

<!--
If this PR makes changes that other people should be aware of when upgrading
their code, describe what they should do between the two UPGRADE INSTRUCTIONS
lines below.
-->
<!-- UPGRADE INSTRUCTIONS -->

<!-- UPGRADE INSTRUCTIONS -->

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc...
-->

knelli2 committed:
Add function to compute coords of horizon iteration

knelli2 committed:
Rename interpolation retries to compute points retries

No interpolation is actually done.

wthrowe commented on src/ParallelAlgorithms/ApparentHorizonFinder/OptionTags.hpp:
```diff
@@ -39,7 +39,7 @@ struct HorizonOptions {
+  struct MaxComputeCoordsRetries {
     static constexpr Options::String help = {
         "Number of times to retry the interpolation where, with each retry, "
         "the two previous surfaces are averaged and that new surface is used."};
```
Do you want to update the help text?

wthrowe commented on src/ParallelAlgorithms/ApparentHorizonFinder/CMakeLists.txt:
```diff
@@ -25,6 +26,7 @@ spectre_target_headers(
   ComputeExcisionBoundaryVolumeQuantities.tpp
   ComputeHorizonVolumeQuantities.hpp
   ComputeHorizonVolumeQuantities.tpp
+  ComputeCoords.hpp
```
'C' < 'H'

wthrowe commented on tests/Unit/ParallelAlgorithms/ApparentHorizonFinder/Test_ComputeCoords.cpp:
```diff
@@ -0,0 +1,246 @@
+
+namespace ah {
+namespace {
+void test_compute_points() {
```
Optional, since I'm not sure how hard it is: It looks like nothing in this test actually tests the extrapolation code, beyond the check that the results were inside the domain.  It would be good to test that.

wthrowe reviewed (changes requested):
Also failing to compile on some of the CI setups.

knelli2 committed:
Add constructor for PreviousSurface

Needed because older compilers don't support parenthesized initialization
of aggregates

knelli2 commented:
Added fixups and a new commit (first commit)

wthrowe commented:
OK.  Squash and rebase.

knelli2 commented:
Squashed and rebased

wthrowe commented:
Something went wrong.

knelli2 commented:
Oops.... forgot a rename
